app/src/main/java/io/homeassistant/companion/android/widgets/BaseWidgetProvider.kt

```diff
@@ -76,8 +76,10 @@ abstract class BaseWidgetProvider : AppWidgetProvider() {
                 updateAllWidgets(context)
                 if (getAllWidgetIds(context).isNotEmpty()) {
                     entityUpdates = integrationUseCase.getEntityUpdates()
-                    entityUpdates!!.collect {
-                        onEntityStateChanged(context, it)
+                    if (entityUpdates != null) {
+                        entityUpdates!!.collect {
+                            onEntityStateChanged(context, it)
+                        }
```

Comment:
Why add a null check + a not-null assertion when you can just use a safe call?
```
entityUpdates?.collect {
    onEntityStateChanged(context, it)
}
```

Reply:
good call :)